Source: Upstream, SVN r864423
Upstream: Yes
Reason: Fix compilation w/ >=dev-libs/openssl-0.9.8g


From 883f1057b31387e7cbfe66f6b3cf5781a6ca49fb Mon Sep 17 00:00:00 2001
From: infiniti <infiniti@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Wed, 24 Sep 2008 19:22:26 +0000
Subject: [PATCH] remove whirlpool usage.  the algorithm is missing on at least 0.9.8g and
 0.9.8i, even though there's an OBJ_whirlpool definition in 0.9.8i.

git-svn-id: svn://anonsvn.kde.org/home/kde/trunk/kdesupport/qca@864423 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 plugins/qca-ossl/qca-ossl.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/qca-ossl.cpp b/qca-ossl.cpp
index 746a863..f9efd84 100644
--- a/qca-ossl.cpp
+++ b/qca-ossl.cpp
@@ -6616,9 +6616,11 @@ static QStringList all_hash_types()
 #ifdef SHA512_DIGEST_LENGTH
 	list += "sha512";
 #endif
+/*
 #ifdef OBJ_whirlpool
 	list += "whirlpool";
 #endif
+*/
 	return list;
 }
 
@@ -6863,10 +6865,12 @@ public:
 		else if ( type == "sha512" )
 			return new opensslHashContext( EVP_sha512(), this, type);
 #endif
+/*
 #ifdef OBJ_whirlpool
 		else if ( type == "whirlpool" )
 			return new opensslHashContext( EVP_whirlpool(), this, type);
 #endif
+*/
 		else if ( type == "pbkdf1(sha1)" )
 			return new opensslPbkdf1Context( EVP_sha1(), this, type );
 		else if ( type == "pbkdf1(md2)" )
-- 
1.6.0.4


From 70730a58fe6d9ba5b50b8971424663ba52947e4c Mon Sep 17 00:00:00 2001
From: infiniti <infiniti@283d02a7-25f6-0310-bc7c-ecb5cbfe19da>
Date: Wed, 24 Sep 2008 19:23:30 +0000
Subject: [PATCH] fix warning

git-svn-id: svn://anonsvn.kde.org/home/kde/trunk/kdesupport/qca@864425 283d02a7-25f6-0310-bc7c-ecb5cbfe19da
---
 plugins/qca-ossl/qca-ossl.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/qca-ossl.cpp b/qca-ossl.cpp
index f9efd84..1790364 100644
--- a/qca-ossl.cpp
+++ b/qca-ossl.cpp
@@ -327,7 +327,7 @@ static X509_EXTENSION *new_subject_key_id(X509 *cert)
 	X509V3_CTX ctx;
 	X509V3_set_ctx_nodb(&ctx);
 	X509V3_set_ctx(&ctx, NULL, cert, NULL, NULL, 0);
-	X509_EXTENSION *ex = X509V3_EXT_conf_nid(NULL, &ctx, NID_subject_key_identifier, "hash");
+	X509_EXTENSION *ex = X509V3_EXT_conf_nid(NULL, &ctx, NID_subject_key_identifier, (char *)"hash");
 	return ex;
 }
 
-- 
1.6.0.4
